openff/toolkit/topology/molecule.py

```diff
@@ -3279,7 +3795,12 @@ def _add_bond_charge_virtual_site(self, atoms, distance, **kwargs):
         vsite = BondChargeVirtualSite(atoms, distance, **kwargs)
 
         self._add_virtual_site(vsite, replace=replace)
-        self._invalidate_cached_properties()
+
+        # If we invalidate, we can never generate virtual sites after
+        # conformers are generated. This seems off. Since virtual sites
+        # do not seem to affect anything cached, we keep it.
+        # self._invalidate_cached_properties()
+
```

Comment:
Agreed. And I'm moving toward thinking that "cached properties" are strictly things that affect the connectivity table, so this change is fine and we can remove this comment. (same goes for the copies of this comment elsewhere)

Reply:
```suggestion
```